// @gluestack-ui/themed-native-base/build/components/AlertDialog/styled-components/Content.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const react_1 = require("@gluestack-style/react");
const animation_resolver_1 = require("@gluestack-style/animation-resolver");
const legend_motion_animation_driver_1 = require("@gluestack-style/legend-motion-animation-driver");
exports.default = (0, react_1.styled)(animation_resolver_1.AnimatedView, {
    // @ts-ignore
    'bg': '$muted.50',
    // @ts-ignore
    'rounded': '$lg',
    'overflow': 'hidden',
    //@ts-ignore
    ':initial': {
        scale: 0.9,
        opacity: 0,
    },
    ':animate': {
        scale: 1,
        opacity: 1,
    },
    ':exit': {
        scale: 0.9,
        opacity: 0,
    },
    ':transition': {
        type: 'spring',
        damping: 18,
        stiffness: 250,
        opacity: {
            type: 'timing',
            duration: 250,
        },
    },
    '_dark': {
        backgroundColor: '$muted.800',
    },
    'defaultProps': {
        shadow: '1',
    },
}, {
    componentName: 'AlertDialogContent',
    ancestorStyle: ['_content'],
}, {
    plugins: [new animation_resolver_1.AnimationResolver(legend_motion_animation_driver_1.MotionAnimationDriver)],
});
